shared/network_ip.go

```diff
@@ -147,10 +147,31 @@ func (r *IPRange) Overlaps(otherRange *IPRange) bool {
 	return r.ContainsIP(otherRange.Start) || r.ContainsIP(otherRange.End)
 }
 
+// OverlapsNetwork checks whether ip range is at least partially inside of a given network.
+func (r *IPRange) OverlapsNetwork(network *net.IPNet) bool {
+	return network.Contains(r.Start) || network.Contains(r.End)
+}
+
 func (r *IPRange) String() string {
 	if r.End == nil {
 		return r.Start.String()
 	}
 
 	return fmt.Sprintf("%v-%v", r.Start, r.End)
 }
+
+// ParseNetworks parses a comma separated list of IP networks in CIDR notation.
+func ParseNetworks(netList string) ([]*net.IPNet, error) {
```

Comment:
Similar to `ParseIPRanges` above, I think we need to trim the space around each network, as the validator uses:

```go
validate.Optional(validate.IsListOf(validate.IsNetworkV6))
```

Which allows for example `10.0.0.0/24,      192.168.0.1/32     , 127.0.0.0/8`

Reply:
Such parser functions make good candidate for quick unit tests. Not asking for this specifically but would be nice to have if you can spare the time ;)